doc/man3/RSA_generate_key.pod

```diff
@@ -2,32 +2,37 @@
 
 =head1 NAME
 
+OSSL_RSA_keygen,
 RSA_generate_key_ex, RSA_generate_key,
 RSA_generate_multi_prime_key - generate RSA key pair
 
 =head1 SYNOPSIS
 
  #include <openssl/rsa.h>
 
+ EVP_PKEY *OSSL_RSA_keygen(OSSL_LIB_CTX *libctx, int bits);
```

Comment:
Would it be interesting to have it also take a `primes`, or to have a second function for multiprime?  So either change to this:


``` C
EVP_PKEY *OSSL_RSA_keygen(OSSL_LIB_CTX *libctx, int bits, int primes);
```
 or add this:

``` C
EVP_PKEY *OSSL_RSA_multiprime_keygen(OSSL_LIB_CTX *libctx, int bits, int primes);
```

Reply:
Yep and this is entirely why I am not sure this is a good idea..
It is a potential API explosion.
I have no problem with the generic one shot OSSL_keygen() though (except for its name) - maybe EVP_PKEY_keygen_something??

Reply:
All this tells me is that people are scared of OSSL_PARAMS - which was supposed to be a replacement for this kind of API bloat (at least I thought it was).

Reply:
> All this tells me is that people are scared of OSSL_PARAMS 

You can't really say there's an opinion about `OSSL_PARAMS` based on this one PR. :)

Reply:
I may form this opinion based on whether we need this API that just sets a single parameter and then calls a one shot function.
Even 'bits' is an optional parameter for keygen since it uses a default if it is not specified..

Reply:
>What about DSA, DH, ED25519, ED448, X25519, X448, SM2?

ED25519, ED448, X25519, X448 are covered easily by
```
    OSSL_keygen(NULL, "ED25519", NULL, NULL);
```
and the like. I've meanwhile made this clear in `EVP_PKEY-X25519.pod`.

I see not much point for special convenience functions for DSA and DH
since there are a lot of variations and these key types have become unpopular.
Neither for for SM2 and the like since they are rarely used.

Reply:
> Would it be interesting to have it also take a primes, or to have a second function for multiprime?

I do not think that it is worth adding more coverage for special forms of RSA key generation etc. because they are pretty rarely used.

Reply:
> It is a potential API explosion.

As mentioned, I believe it is worth just adding rather simple `EVP_RSA_keygen()` and `EVP_EC_keygen()` convenience functions because they already cover the commonly used cases.

> I have no problem with the generic one shot OSSL_keygen() though (except for its name) - maybe EVP_PKEY_keygen_something??

It's meanwhile `EVP_keygen()`. 
As mentioned, `EVP_PKEY_keygen()` is already taken.

Reply:
> All this tells me is that people are scared of OSSL_PARAMS

I would not be afraid of using them, 
but for common cases I consider them plainly inconvenient to use.